cmd/podman/trust.go

```diff
@@ -0,0 +1,293 @@
+		Action:       setTrustCmd,
+		OnUsageError: usageErrorHandler,
+	}
+
+	showTrustDescription = "Display trust policy for the system"
+	showTrustCommand     = cli.Command{
+		Name:                   "show",
+		Usage:                  "Display trust policy for the system",
+		Description:            showTrustDescription,
+		Flags:                  sortFlags(showTrustFlags),
+		Action:                 showTrustCmd,
+		ArgsUsage:              "",
+		UseShortOptionHandling: true,
+		OnUsageError:           usageErrorHandler,
+	}
+
+	trustSubCommands = []cli.Command{
+		setTrustCommand,
+		showTrustCommand,
+	}
+
+	trustDescription = fmt.Sprintf(`Manages the trust policy of the host system. (%s)
+	 Trust policy describes a registry scope that must be signed by public keys.`, getDefaultPolicyPath())
+	trustCommand = cli.Command{
+		Name:         "trust",
+		Usage:        "Manage container image trust policy",
+		Description:  trustDescription,
+		ArgsUsage:    "{set,show} ...",
+		Subcommands:  trustSubCommands,
+		OnUsageError: usageErrorHandler,
+	}
+)
+
+func showTrustCmd(c *cli.Context) error {
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not create runtime")
+	}
+
+	var (
+		policyPath              string
+		systemRegistriesDirPath string
+	)
+	if c.IsSet("policypath") {
+		policyPath = c.String("policypath")
+	} else {
+		policyPath = trust.DefaultPolicyPath(runtime.SystemContext())
+	}
+	policyContent, err := ioutil.ReadFile(policyPath)
+	if err != nil {
+		return errors.Wrapf(err, "unable to read %s", policyPath)
+	}
+	if c.IsSet("registrypath") {
+		systemRegistriesDirPath = c.String("registrypath")
+	} else {
+		systemRegistriesDirPath = trust.RegistriesDirPath(runtime.SystemContext())
+	}
+
+	if c.Bool("raw") {
+		_, err := os.Stdout.Write(policyContent)
+		if err != nil {
+			return errors.Wrap(err, "could not read trust policies")
+		}
+		return nil
+	}
+
+	var policyContentStruct trust.PolicyContent
+	if err := json.Unmarshal(policyContent, &policyContentStruct); err != nil {
+		return errors.Errorf("could not read trust policies")
+	}
+	policyJSON, err := trust.GetPolicyJSON(policyContentStruct, systemRegistriesDirPath)
+	if err != nil {
+		return errors.Wrapf(err, "error reading registry config file")
+	}
+	if c.Bool("json") {
+		var outjson interface{}
+		outjson = policyJSON
+		out := formats.JSONStruct{Output: outjson}
+		return formats.Writer(out).Out()
+	}
+
+	sortedRepos := sortPolicyJSONKey(policyJSON)
+	type policydefault struct {
+		Repo      string
+		Trusttype string
+		GPGid     string
+		Sigstore  string
+	}
+	var policyoutput []policydefault
+	for _, repo := range sortedRepos {
+		repoval := policyJSON[repo]
+		var defaultstruct policydefault
+		defaultstruct.Repo = repo
+		if repoval["type"] != nil {
+			defaultstruct.Trusttype = trustTypeDescription(repoval["type"].(string))
+		}
+		if repoval["keys"] != nil && len(repoval["keys"].([]string)) > 0 {
```

Comment:
Code like this strongly suggests that `GetPolicyJSON` should not be producing output in the `--json` format directly; instead, the `trust` subpackage should return well-typed data in Go structs conceptually independent form the presentation format (maybe much closer to the `policydefault` data type, or just the original `trust.RepoContent`), and this function should have two different formatters, one into `--json`, one into the other format as `[]policyoutput`.